File: day3/first.py
import re
result = 0
with open("input.txt") as file:
    lines = file.read().splitlines()
    for line in lines:
        l = list(line)
        first_half, second_half = l[:len(l)//2], l[len(l)//2:]
        priority_char = list(set(first_half).intersection(set(second_half)))[0]
        if re.match(r'^[a-z]$', priority_char):
            result += ord(priority_char) - 96  # 1 is the first value
        else:
            result += ord(priority_char) - 65 + 27
    print(result)

File: day3/first.rb
require 'set'
result = 0
File.open("input.txt") do |file|
  content = file.read()
  lines = content.split("\n")
  lines.each do |line|
    first_half, second_half = line.chars.each_slice(line.chars.size / 2).to_a
    priority_char = first_half.to_set.intersection(second_half.to_set).to_a[0]
    if ('a'..'z').include?(priority_char)
      result += priority_char.ord - 96
    else
      result += priority_char.ord - 65 + 27
    end
  end
  puts result
end
